### PR TITLE
Dls 5096 gaurav

### PR DESCRIPTION
**Pull request by daglisood**

### CODE REVIEW

**Comment by priyankadhiman3006** [comment]
Since this branch has all the changes , it's very difficult to review. Will it okay for you to wait till our branches get merged and then we review it?

**Comment by daglisood** [comment]
Since its only going into the ui-upgrade-base, we should be able to review it. It wouldn't make sense for this ticket to wait for in progress tickets. When other tickets are ready for review it would make their review quicker

**Comment by priyankadhiman3006** [comment]
> Since its only going into the ui-upgrade-base, we should be able to review it. It wouldn't make sense for this ticket to wait for in progress tickets. When other tickets are ready for review it would make their review quicker

My point is , there is alot in this PR which has already been reviewed and it's difficult to differentiate what are the new changes

File: app/views/inflation_proof.scala.html [hunk 1]
```diff
@@ -29,84 +32,72 @@
...
+            legend = Some(Legend(
+                content = Text(Messages("gmp.inflation_proofed")),
+                classes = "govuk-fieldset__legend--m",
+                isPageHeading = true
+            ))
+        )),
+        items = Seq(
+            InputItem(
+                name = "revaluationDate.day",
+                classes = "govuk-input--width-2",
```
Comment (priyankadhiman3006) [inline app/views/inflation_proof.scala.html:51]: Error highlight is missing from field if there is error in there

File: app/views/inflation_proof.scala.html [hunk 2]
```diff
@@ -29,84 +32,72 @@
...
 @gmpMain(title = Messages("gmp.inflation_proof.question") + " - " + Messages("service.title") + " - " + Messages("gov.uk"), pageScripts = Some(pageScripts)) {
 
 
     @includes.back_link(Messages("gmp.inflation_proof.question"))
 
-    @views.html.helpers.errorSummary(form = inflationProofForm, focusId = {if(!inflationProofForm.hasGlobalErrors && inflationProofForm.error("revaluate").isDefined) Some("revaluate-revaluation-date") else Some("revaluationDate_day")})
 
         @viewHelpers.form(action = routes.InflationProofController.post) {
+            @if(inflationProofForm.errors.nonEmpty){
+                @govukErrorSummary(ErrorSummary().withFormErrorsAsText(inflationProofForm))
```
Comment (priyankadhiman3006) [inline app/views/inflation_proof.scala.html:77]: We have to print individual errors as per new dat component for example if day is missing, the error should say day is missing and highlight the day field in red. if month and year are missing , then it should say the same and highlight those fields in red

File: app/views/inflation_proof.scala.html [hunk 3]
```diff
@@ -29,84 +32,72 @@
...
+<hr class="govuk-section-break govuk-section-break--m" aria-hidden = true />
+<p class="govuk-body">@Messages("gmp.inflationproof.subtext")</p>
+}
+
+
 
 @gmpMain(title = Messages("gmp.inflation_proof.question") + " - " + Messages("service.title") + " - " + Messages("gov.uk"), pageScripts = Some(pageScripts)) {
 
 
     @includes.back_link(Messages("gmp.inflation_proof.question"))
```
Comment (priyankadhiman3006) [inline app/views/inflation_proof.scala.html:72]: update the back link as
```
  @viewHelpers.govukBackLink(BackLink(
        href = controllers.routes.InflationProofController.back.url,
        content = Text("Back")
    ))
```

**Comment by priyankadhiman3006** [comment]
Please do rebase with DLS_5489 (ginny's branch as it has my latest changes from 4546)

**Comment by daglisood** [comment]
Not required

**Pull request closed without merging**